src/app/community-view/community-view.component.ts

```diff
@@ -121,6 +122,32 @@ export class CommunityViewComponent implements OnInit {
     });
   }
 
+  canChangeRole(user: User): boolean {
+    return this.isCommunityManager && user.id !== this.currentUserId;
+  }
+
+  changeRole(user: User, isRaising: boolean) {
```

Comment:
To tell the truth I do not support any boolean flags as arguments of business methods. I would recommend to create two methods like `promote(user: User)` and `demote(user: User)` and change the access modifier of `changeRole` to `private`. `promote(user: User)` can call `changeRole(user, true)` internally and `demote(user: User)` can call `changeRole(user, false)`. It would enable the component to be more business-oriented. What do you think?

Reply:
If you go for the approach with two business methods like `promoteToManager` and `demoteToMember` then I think those methods can call the internal, private method `changeRole` with the appropriate `CommunityRole` enum value as an argument, not with `true` or `false`.

Reply:
Georg, Dmitry, thank you. done